identity-account/src/identity/chain_state.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::Serialize;
+
+use identity_iota::tangle::MessageId;
+use identity_iota::tangle::MessageIdExt;
+
+/// Holds the last published message ids of the integration and diff chains.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct ChainState {
+  #[serde(default = "MessageId::null", skip_serializing_if = "MessageId::is_null")]
+  previous_integration_message_id: MessageId,
+  #[serde(default = "MessageId::null", skip_serializing_if = "MessageId::is_null")]
+  previous_diff_message_id: MessageId,
+}
+
+impl ChainState {
+  pub fn new() -> Self {
+    Self {
+      previous_integration_message_id: MessageId::null(),
+      previous_diff_message_id: MessageId::null(),
+    }
+  }
+
+  /// Returns the previous integration message id of the identity.
+  pub fn previous_integration_message_id(&self) -> &MessageId {
+    &self.previous_integration_message_id
+  }
+
+  /// Returns the previous diff message id, or the previous integration message id.
+  pub fn diff_message_id(&self) -> &MessageId {
+    if self.previous_diff_message_id.is_null() {
+      &self.previous_integration_message_id
+    } else {
+      &self.previous_diff_message_id
+    }
+  }
```

Comment:
Nit: it might be better to return `Option<&MessageId>` and move the logic where it defaults to the `previous_integration_message_id` to where it's used?

Reply:
Good suggestion! I would still use a bare `MessageId`, though, since that has a `null` variant in itself. With `Option<MessageId>` you'd still have to technically check whether it is null. Since it's only used internally, that seems fine.

Reply:
You're absolutely right, I forgot that `MessageId` needs to be checked for `null` regardless so returning `&MessageId` is absolutely fine. Might need to note in the doc comment here and in `previous_integration_message_id` that it could be `null`.